sql/sqlwire/wire.proto

```diff
@@ -102,6 +100,16 @@ message Request {
   repeated Datum params = 3 [(gogoproto.nullable) = false];
 }
 
+// ResponseWithError is what we send on the wire. It includes the error so that
+// Response doesn't need to, simplifying error handling that involves handling
+// these objects.
+message ResponseWithError {
```

Comment:
Feels a bit strange to have `ResponseWithError` when there is only a single `sqlwire.Response` type. You could easily leave the protos here untouched while keeping the same changes to the rest of the code (i.e. returning an `error` from various functions).